### PR TITLE
Login consistency

### PR DESCRIPTION
**Pull request by yhaspel**
Logout tab added to side menu.
Logout on tab change when no accounts in props.
Logout on refresh app.

### CODE REVIEW

File: app/basicComponents/SmInput.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ const Wrapper = styled.div`
 
 // $FlowStyledIssue
 const Input = styled.input`
-  width: 100%;
```
Comment (IlyaVi) [inline app/basicComponents/SmInput.js]: what's the reason to remove it?

File: app/screens/main/Main.js [hunk 2]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
     const isSameLocation = !!newPath && window.location.hash.endsWith(newPath);
     if (!isSameLocation && newPath) {
       this.setState({ selectedItemIndex: index });
       history.push(newPath);
     }
   };
+
+  logoutToAuth = () => {
+    const { history, logout } = this.props;
+    logout();
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: first push history and then do logout

File: app/screens/main/Main.js [hunk 3]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
+      return;
+    }
     const isSameLocation = !!newPath && window.location.hash.endsWith(newPath);
     if (!isSameLocation && newPath) {
       this.setState({ selectedItemIndex: index });
       history.push(newPath);
     }
   };
+
+  logoutToAuth = () => {
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: give this proper name

File: app/screens/main/Main.js [hunk 4]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
     );
   }
 
+  componentDidUpdate(prevProps, prevState) {
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: what the scenario that can cause this?

File: app/basicComponents/SmInput.js (hunk 1 above)
Comment (yhaspel) [reply app/basicComponents/SmInput.js]: width 100% is defined in Wrapper and this inherits the correct width. Setting 100% width here overflows the width (can be seen in any of the steps container pages).

File: app/screens/main/Main.js (hunk 2 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: ok

File: app/screens/main/Main.js (hunk 4 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: Refresh causes this.

File: app/screens/main/Main.js (hunk 3 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: ok

File: app/basicComponents/SmInput.js (hunk 1 above)
Comment (IlyaVi) [reply app/basicComponents/SmInput.js]: did you check the latest dev version?

File: app/screens/main/Main.js (hunk 4 above)
Comment (IlyaVi) [reply app/screens/main/Main.js]: which refresh?

File: app/screens/main/Main.js [hunk 5]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
+
   handleSideMenuPress = ({ index }: { index: number }) => {
-    const { history } = this.props;
-    const newPath = sideMenuItems[index].path;
+    const { history, accounts } = this.props;
+    const newPath: ?string = sideMenuItems[index].path;
+    const isLocationLocalNode = newPath && newPath.includes('/local-node');
+    if (!(accounts || isLocationLocalNode) || newPath === '/') {
+      this.navToAuthAndLogout();
+      return;
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: rewrite it without using empty return

File: app/screens/main/Main.js [hunk 6]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
+    }
+  }
+
   handleSideMenuPress = ({ index }: { index: number }) => {
-    const { history } = this.props;
-    const newPath = sideMenuItems[index].path;
+    const { history, accounts } = this.props;
+    const newPath: ?string = sideMenuItems[index].path;
+    const isLocationLocalNode = newPath && newPath.includes('/local-node');
+    if (!(accounts || isLocationLocalNode) || newPath === '/') {
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: (!accounts && !isLocationLocalNode) || newPath === '/'  is much more readable

File: app/screens/main/Main.js [hunk 7]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
+      this.navToAuthAndLogout();
+    }
+  }
+
   handleSideMenuPress = ({ index }: { index: number }) => {
-    const { history } = this.props;
-    const newPath = sideMenuItems[index].path;
+    const { history, accounts } = this.props;
+    const newPath: ?string = sideMenuItems[index].path;
+    const isLocationLocalNode = newPath && newPath.includes('/local-node');
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: give more meaningful name

File: app/screens/main/Main.js [hunk 8]
```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
...
+    if (!(accounts || isLocationLocalNode) || newPath === '/') {
+      this.navToAuthAndLogout();
+      return;
+    }
     const isSameLocation = !!newPath && window.location.hash.endsWith(newPath);
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: don't use window object reference

File: app/screens/main/Main.js (hunk 4 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: Refreshing the app.

File: app/screens/main/Main.js (hunk 6 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: OK.

File: app/screens/main/Main.js (hunk 7 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: ok

File: app/screens/main/Main.js (hunk 8 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: will be changed.

File: app/screens/main/Main.js (hunk 5 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: ok

File: app/basicComponents/SmInput.js (hunk 1 above)
Comment (yhaspel) [reply app/basicComponents/SmInput.js]: pulled and merged with dev. This change was following that.

Comment (IlyaVi) [reply app/basicComponents/SmInput.js]: please attach screenshot with overflowed element caused by width: 100%

File: app/screens/main/Main.js (hunk 4 above)
Comment (IlyaVi) [reply app/screens/main/Main.js]: we are using MemoryRouter now, every reload will cause landing at top route level no matter what caused it. Does this scenario you described still exist?

File: app/basicComponents/SmInput.js (hunk 1 above)
Comment (yhaspel) [reply app/basicComponents/SmInput.js]: I don't see this issue anymore. With or without width: 100%.

File: app/screens/main/Main.js (hunk 4 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: I'll recheck and remove if so.

**Pull request merged**